Src/FluentAssertions/Equivalency/SelfReferenceEquivalencyAssertionOptions.cs

```diff
@@ -62,12 +56,12 @@ public abstract class SelfReferenceEquivalencyAssertionOptions<TSelf> : IEquival
     private bool ignoreNonBrowsableOnSubject;
     private bool excludeNonBrowsableOnExpectation;
 
-    private bool? compareRecordsByValue;
-
     #endregion
 
     private protected SelfReferenceEquivalencyAssertionOptions()
     {
+        equalityStrategyCache = new EqualityStrategyCache(null);
```

Comment:
🔧 Instead of passing a magic `null` here, I think we should just have a parameterless constructor.

Reply:
Fixed in [commit 8249eea](https://github.com/fluentassertions/fluentassertions/pull/2431/commits/8249eea1fb65ce69c8ecd533ddca4ffb437488c3).